crates/bevy_reflect/derive/src/from_reflect.rs

```diff
@@ -328,3 +390,35 @@ fn get_field_accessor(field: &Field, index: usize, is_tuple: bool) -> Lit {
             .unwrap_or_else(|| Lit::Str(LitStr::new(&index.to_string(), Span::call_site())))
     }
 }
+
+fn get_conversions(meta: &ReflectMeta) -> proc_macro2::TokenStream {
+    let bevy_reflect_path = meta.bevy_reflect_path();
+    let conversions: Vec<_> = meta
+        .attrs()
+        .conversions()
+        .iter()
+        .map(|conversion| {
+            let path = &conversion.path;
+            match &conversion.func {
+                Some(func) => quote! {
+                    if let #FQOption::Some(value) =
+                    <#path as #bevy_reflect_path::FromReflect>::from_reflect(reflect) {
+                        #FQOption::Some((#func)(value).into())
+                    }
+                },
+                None => quote! {
+                    if let #FQOption::Some(value) =
+                    <#path as #bevy_reflect_path::FromReflect>::from_reflect(reflect) {
+                        #FQOption::Some(value.into())
+                    }
+                },
+            }
+        })
+        .collect();
+
+    if conversions.is_empty() {
+        quote! {}
+    } else {
+        quote! { else #(#conversions)else* }
```

Comment:
Nit: I wonder if, for simplicity, we could just have `conversions` be a series of if statements that return. Then we shouldn't need to wrap these in els-if statements.

You might also be able to do that for fields too.